admin-api/authorization_server/external_authorization.md

```diff
@@ -0,0 +1,161 @@
+
+Clone the module repository into your `modules` folder, then go to the Module manager in your back office and install the `Keycloak OAuth2 connector demo` module.
+
+Then go to the configuration and provide the two needed urls for configuration:
+
+- **Keycloak realm endpoint**: the url on which your server can access the keycloak realm endpoint, this is the base realm URL from which the connector can fetch the certificates and validate if the JWT Token is valid
+- **Keycloak allowed issuers**: A list of issuer values that are considered valid, usually it's the same as the realm endpoint But in case you request your access token from different urls you can specify multiple endpoints so the validation process accepts multiple issuers.
+
+If you're using the included docker locally then the default configuration should be enough:
+
+![Keycloak configuration](../img/keycloak_configuration.png)
+
+{{% notice note %}}
+For security reason the values in this form are encrypted in the database.
+{{% /notice %}}
+
+## Authorization server implementation
+
+You can see the [exact implementation of the authorization server](https://github.com/PrestaShop/keycloak_connector_demo/blob/v1.1.0/src/OAuth2/KeycloakAuthorizationServer.php) in the module itself but here is an explanation about chat each methods does:
+
+```php
+namespace PrestaShop\Module\KeycloakConnectorDemo\OAuth2;
+
+class KeycloakAuthorizationServer implements AuthorisationServerInterface
+{
+    public function isTokenValid(Request $request): bool
+    {
+        // Parses the JWT Token and check if it's valid
+        // Fetch the list of allowed issuers from the configuration, if the Token issuer matches one of the allowed ones
+        // Fetch the URL realm from the configuration (ex: http://localhost:8003/realms/prestashop)
+        // Download the certificates from the authorization server (ex: http://localhost:8003/realms/prestashop/protocol/openid-connect/certs)
+        // Check if the JWT token was correctly signed based on the public certificate
+        // If all these checks passed then the token is valid and issued by Keycloak and the method returns true
+
+        // Any failure along this test the method returns false
+    }
+
+    public function getJwtTokenUser(Request $request): ?JwtTokenUser
+    {
+        // Parses the JWT token from the request, it should contain these claims
+        //  - clientId: The used client ID to get the access token
+        //  - scope: a list of scope separated by spaces
+        //  - iss: the issuer that granted the access token (ex: http://localhost:8003/realms/prestashop)
+        // With these three data the method can return a JwtTokenUser DTO
+    }
+}
+```
+
+Once your service is implemented you just need to define in your module's services, PrestaShop will automatically scan and detect and it will be used in the loop that checks for authorization servers in each request of the resource server.
```

Comment:
```suggestion
Once your service is implemented, you just need to define in your module's services. PrestaShop will automatically scan and detect it and it will be used in the loop that checks for authorization servers in each request of the resource server.
```